File: binding/src/libqgraphicslayout.cpp
#include <QEvent>
#include <QGraphicsLayout>
#include <QGraphicsLayoutItem>
#include <qgraphicslayout.h>
#include "libqgraphicslayout.h"
#include "libqgraphicslayout.hxx"

QGraphicsLayout* QGraphicsLayout_new() {
	 return new VirtualQGraphicsLayout();
}

QGraphicsLayout* QGraphicsLayout_new2(QGraphicsLayoutItem* parent) {
	 return new VirtualQGraphicsLayout(parent);
}

void QGraphicsLayout_SetContentsMargins(QGraphicsLayout* self, double left, double top, double right, double bottom) {
	self->setContentsMargins(left, top, right, bottom);
}

void QGraphicsLayout_GetContentsMargins(const QGraphicsLayout* self, double* left, double* top, double* right, double* bottom) {
	self->getContentsMargins(left, top, right, bottom);
}

void QGraphicsLayout_Activate(QGraphicsLayout* self) {
	self->activate();
}

bool QGraphicsLayout_IsActivated(const QGraphicsLayout* self) {
	return self->isActivated();
}

void QGraphicsLayout_Invalidate(QGraphicsLayout* self) {
	self->invalidate();
}

void QGraphicsLayout_UpdateGeometry(QGraphicsLayout* self) {
	self->updateGeometry();
}

void QGraphicsLayout_WidgetEvent(QGraphicsLayout* self, QEvent* e) {
	self->widgetEvent(e);
}

int QGraphicsLayout_Count(const QGraphicsLayout* self) {
	return self->count();
}

QGraphicsLayoutItem* QGraphicsLayout_ItemAt(const QGraphicsLayout* self, int i) {
	return self->itemAt(i);
}

void QGraphicsLayout_RemoveAt(QGraphicsLayout* self, int index) {
	self->removeAt(index);
}

void QGraphicsLayout_SetInstantInvalidatePropagation(bool enable) {
	QGraphicsLayout::setInstantInvalidatePropagation(enable);
}

bool QGraphicsLayout_InstantInvalidatePropagation() {
	return QGraphicsLayout::instantInvalidatePropagation();
}

// Base class handler implementation
void QGraphicsLayout_QBaseGetContentsMargins(const QGraphicsLayout* self, double* left, double* top, double* right, double* bottom) {
	auto* vqgraphicslayout = dynamic_cast<const VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_GetContentsMargins_IsBase(true);
	vqgraphicslayout->getContentsMargins(left, top, right, bottom);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLayout_OnGetContentsMargins(const QGraphicsLayout* self, intptr_t slot) {
	auto* vqgraphicslayout = dynamic_cast<const VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_GetContentsMargins_Callback(reinterpret_cast<VirtualQGraphicsLayout::QGraphicsLayout_GetContentsMargins_Callback>(slot));
}
}

// Base class handler implementation
void QGraphicsLayout_QBaseInvalidate(QGraphicsLayout* self) {
	auto* vqgraphicslayout = dynamic_cast<VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_Invalidate_IsBase(true);
	vqgraphicslayout->invalidate();
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLayout_OnInvalidate(QGraphicsLayout* self, intptr_t slot) {
	auto* vqgraphicslayout = dynamic_cast<VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_Invalidate_Callback(reinterpret_cast<VirtualQGraphicsLayout::QGraphicsLayout_Invalidate_Callback>(slot));
}
}

// Base class handler implementation
void QGraphicsLayout_QBaseUpdateGeometry(QGraphicsLayout* self) {
	auto* vqgraphicslayout = dynamic_cast<VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_UpdateGeometry_IsBase(true);
	vqgraphicslayout->updateGeometry();
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLayout_OnUpdateGeometry(QGraphicsLayout* self, intptr_t slot) {
	auto* vqgraphicslayout = dynamic_cast<VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_UpdateGeometry_Callback(reinterpret_cast<VirtualQGraphicsLayout::QGraphicsLayout_UpdateGeometry_Callback>(slot));
}
}

// Base class handler implementation
void QGraphicsLayout_QBaseWidgetEvent(QGraphicsLayout* self, QEvent* e) {
	auto* vqgraphicslayout = dynamic_cast<VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_WidgetEvent_IsBase(true);
	vqgraphicslayout->widgetEvent(e);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLayout_OnWidgetEvent(QGraphicsLayout* self, intptr_t slot) {
	auto* vqgraphicslayout = dynamic_cast<VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_WidgetEvent_Callback(reinterpret_cast<VirtualQGraphicsLayout::QGraphicsLayout_WidgetEvent_Callback>(slot));
}
}

// Base class handler implementation
int QGraphicsLayout_QBaseCount(const QGraphicsLayout* self) {
	auto* vqgraphicslayout = dynamic_cast<const VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_Count_IsBase(true);
	return vqgraphicslayout->count();
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLayout_OnCount(const QGraphicsLayout* self, intptr_t slot) {
	auto* vqgraphicslayout = dynamic_cast<const VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_Count_Callback(reinterpret_cast<VirtualQGraphicsLayout::QGraphicsLayout_Count_Callback>(slot));
}
}

// Base class handler implementation
QGraphicsLayoutItem* QGraphicsLayout_QBaseItemAt(const QGraphicsLayout* self, int i) {
	auto* vqgraphicslayout = dynamic_cast<const VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_ItemAt_IsBase(true);
	return vqgraphicslayout->itemAt(i);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLayout_OnItemAt(const QGraphicsLayout* self, intptr_t slot) {
	auto* vqgraphicslayout = dynamic_cast<const VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_ItemAt_Callback(reinterpret_cast<VirtualQGraphicsLayout::QGraphicsLayout_ItemAt_Callback>(slot));
}
}

// Base class handler implementation
void QGraphicsLayout_QBaseRemoveAt(QGraphicsLayout* self, int index) {
	auto* vqgraphicslayout = dynamic_cast<VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_RemoveAt_IsBase(true);
	vqgraphicslayout->removeAt(index);
}
}

// Auxiliary method to allow providing re-implementation
void QGraphicsLayout_OnRemoveAt(QGraphicsLayout* self, intptr_t slot) {
	auto* vqgraphicslayout = dynamic_cast<VirtualQGraphicsLayout*>(self);
	if (vqgraphicslayout && vqgraphicslayout->isVirtualQGraphicsLayout) {
vqgraphicslayout->setQGraphicsLayout_RemoveAt_Callback(reinterpret_cast<VirtualQGraphicsLayout::QGraphicsLayout_RemoveAt_Callback>(slot));
}
}

void QGraphicsLayout_Delete(QGraphicsLayout* self) {
    delete self;
}

File: binding/src/libqgraphicslayout.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");%%_IMPORTLIBS_%%
%%_STRUCTDEFS_%%

/// https://doc.qt.io/qt-6/qgraphicslayout.html
pub const qgraphicslayout = struct {

    /// New constructs a new QGraphicsLayout object.
    pub fn New() QtC.QGraphicsLayout {
        return qtc.QGraphicsLayout_new();
    }


    /// New2 constructs a new QGraphicsLayout object.
    pub fn New2(parent: ?*anyopaque) QtC.QGraphicsLayout {
        return qtc.QGraphicsLayout_new2(@ptrCast(parent));
    }


    pub fn SetContentsMargins(self: ?*anyopaque, left: f64, top: f64, right: f64, bottom: f64) void {
        qtc.QGraphicsLayout_SetContentsMargins(@ptrCast(self), left, top, right, bottom);
    }

    pub fn GetContentsMargins(self: ?*anyopaque, left: *f64, top: *f64, right: *f64, bottom: *f64) void {
        qtc.QGraphicsLayout_GetContentsMargins(@ptrCast(self), @ptrCast(left), @ptrCast(top), @ptrCast(right), @ptrCast(bottom));
    }

    /// Allows for overriding the related default method
    pub fn OnGetContentsMargins(self: ?*anyopaque, callback: *const fn (?*anyopaque, *f64, *f64, *f64, *f64) callconv(.c) void) void {
        qtc.QGraphicsLayout_OnGetContentsMargins(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseGetContentsMargins(self: ?*anyopaque, left: *f64, top: *f64, right: *f64, bottom: *f64) void {
        qtc.QGraphicsLayout_QBaseGetContentsMargins(@ptrCast(self), @ptrCast(left), @ptrCast(top), @ptrCast(right), @ptrCast(bottom));
    }

    pub fn Activate(self: ?*anyopaque, ) void {
        qtc.QGraphicsLayout_Activate(@ptrCast(self));
    }

    pub fn IsActivated(self: ?*anyopaque, ) bool {
        return qtc.QGraphicsLayout_IsActivated(@ptrCast(self));
    }

    pub fn Invalidate(self: ?*anyopaque, ) void {
        qtc.QGraphicsLayout_Invalidate(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnInvalidate(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsLayout_OnInvalidate(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseInvalidate(self: ?*anyopaque, ) void {
        qtc.QGraphicsLayout_QBaseInvalidate(@ptrCast(self));
    }

    pub fn UpdateGeometry(self: ?*anyopaque, ) void {
        qtc.QGraphicsLayout_UpdateGeometry(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnUpdateGeometry(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsLayout_OnUpdateGeometry(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseUpdateGeometry(self: ?*anyopaque, ) void {
        qtc.QGraphicsLayout_QBaseUpdateGeometry(@ptrCast(self));
    }

    pub fn WidgetEvent(self: ?*anyopaque, e: ?*anyopaque) void {
        qtc.QGraphicsLayout_WidgetEvent(@ptrCast(self), @ptrCast(e));
    }

    /// Allows for overriding the related default method
    pub fn OnWidgetEvent(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsLayout_OnWidgetEvent(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseWidgetEvent(self: ?*anyopaque, e: ?*anyopaque) void {
        qtc.QGraphicsLayout_QBaseWidgetEvent(@ptrCast(self), @ptrCast(e));
    }

    pub fn Count(self: ?*anyopaque, ) i32 {
        return qtc.QGraphicsLayout_Count(@ptrCast(self));
    }

    /// Allows for overriding the related default method
    pub fn OnCount(self: ?*anyopaque, callback: *const fn (?*anyopaque) callconv(.c) i32) void {
        qtc.QGraphicsLayout_OnCount(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseCount(self: ?*anyopaque, ) i32 {
        return qtc.QGraphicsLayout_QBaseCount(@ptrCast(self));
    }

    pub fn ItemAt(self: ?*anyopaque, i: i32) QtC.QGraphicsLayoutItem {
        return qtc.QGraphicsLayout_ItemAt(@ptrCast(self), i);
    }

    /// Allows for overriding the related default method
    pub fn OnItemAt(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) QtC.QGraphicsLayoutItem) void {
        qtc.QGraphicsLayout_OnItemAt(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseItemAt(self: ?*anyopaque, i: i32) QtC.QGraphicsLayoutItem {
        return qtc.QGraphicsLayout_QBaseItemAt(@ptrCast(self), i);
    }

    pub fn RemoveAt(self: ?*anyopaque, index: i32) void {
        qtc.QGraphicsLayout_RemoveAt(@ptrCast(self), index);
    }

    /// Allows for overriding the related default method
    pub fn OnRemoveAt(self: ?*anyopaque, callback: *const fn (?*anyopaque, i32) callconv(.c) void) void {
        qtc.QGraphicsLayout_OnRemoveAt(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseRemoveAt(self: ?*anyopaque, index: i32) void {
        qtc.QGraphicsLayout_QBaseRemoveAt(@ptrCast(self), index);
    }

    pub fn SetInstantInvalidatePropagation(enable: bool) void {
        qtc.QGraphicsLayout_SetInstantInvalidatePropagation(enable);
    }

    pub fn InstantInvalidatePropagation() bool {
        return qtc.QGraphicsLayout_InstantInvalidatePropagation();
    }

    pub fn AddChildLayoutItem(self: ?*anyopaque, layoutItem: ?*anyopaque) void {
        qtc.QGraphicsLayout_AddChildLayoutItem(@ptrCast(self), @ptrCast(layoutItem));
    }

    /// Allows for overriding the related default method
    pub fn OnAddChildLayoutItem(self: ?*anyopaque, callback: *const fn (?*anyopaque, ?*anyopaque) callconv(.c) void) void {
        qtc.QGraphicsLayout_OnAddChildLayoutItem(@ptrCast(self), @intCast(@intFromPtr(callback)));
    }

    /// Base class method implementation
    pub fn QBaseAddChildLayoutItem(self: ?*anyopaque, layoutItem: ?*anyopaque) void {
        qtc.QGraphicsLayout_QBaseAddChildLayoutItem(@ptrCast(self), @ptrCast(layoutItem));
    }

    /// Delete this object from C++ memory.
    pub fn QDelete(self: ?*anyopaque) void {
        qtc.QGraphicsLayout_Delete(@ptrCast(self));
    }
};
